electrolyzer/simulation/stack.py

```diff
@@ -194,34 +213,107 @@ def __attrs_post_init__(self) -> None:
             ]
         )
 
-        # self.h2_pres_out = 31  # H2 outlet pressure (bar)
-
         self.DTSS = self.calc_state_space()
+        self.d_eol = self.calc_end_of_life_voltage()
 
-    def run(self, P_in):
+    def run_power_deg_penalty(self, P_in):
+        """Run the stack with a degradation penalty applied to the power consumption.
+        Power consumed may be greater than power input.
+
+        Args:
+            P_in (float): stack power input in Wdc
+
+        Returns:
+            2-element tuple containing
+
+            - **I_stack** (float): stack current in Amps
+            - **V_cell** (float): cell voltage in Volts
+        """
+
+        I_stack = self.electrolyzer_model(
```

Comment:
I'm still a bit confused how this applies a penalty on the power consumed. Maybe we can chat about it! Not a blocker, just to better understand the code.